core/Enclave/enclave_self_test.cpp

```diff
@@ -0,0 +1,809 @@
+    (void)sm4_ctr_encrypt(key, _ciphertext, plaintext, VECTOR_LENGTH("plaintext"), iv);
+
+    return TEST_COMPARE(ciphertext);
+}
+
+static bool sm4_ctr_decryption(map<string, string> test_vector)
+{
+    GET_PARAMETER(key);
+    GET_PARAMETER(plaintext);
+    GET_PARAMETER(iv);
+    GET_PARAMETER(ciphertext);
+
+    uint8_t _plaintext[VECTOR_LENGTH("plaintext")] = {0};
+    (void)sm4_ctr_decrypt(key, _plaintext, ciphertext, VECTOR_LENGTH("plaintext"), iv);
+
+    return TEST_COMPARE(plaintext);
+}
+
+static bool sm4_cbc_encryption(map<string, string> test_vector)
+{
+    GET_PARAMETER(key);
+    GET_PARAMETER(plaintext);
+    GET_PARAMETER(iv);
+    GET_PARAMETER(ciphertext);
+
+    uint8_t _ciphertext[VECTOR_LENGTH("plaintext")] = {0};
+    (void)sm4_cbc_encrypt(key, _ciphertext, plaintext, VECTOR_LENGTH("plaintext"), iv);
+
+    return TEST_COMPARE(ciphertext);
+}
+
+static bool sm4_cbc_decryption(map<string, string> test_vector)
+{
+    GET_PARAMETER(key);
+    GET_PARAMETER(plaintext);
+    GET_PARAMETER(iv);
+    GET_PARAMETER(ciphertext);
+
+    uint8_t _plaintext[VECTOR_LENGTH("plaintext")] = {0};
+    (void)sm4_cbc_decrypt(key, _plaintext, ciphertext, VECTOR_LENGTH("plaintext") + VECTOR_LENGTH("iv"), iv);
+
+    return TEST_COMPARE(plaintext);
+}
+
+static bool rsa_crypto(map<string, string> test_vector)
+{
+    GET_PARAMETER(n);
+    GET_PARAMETER(e);
+    GET_PARAMETER(d);
+    GET_PARAMETER(p);
+    GET_PARAMETER(q);
+    GET_PARAMETER(dmp1);
+    GET_PARAMETER(dmq1);
+    GET_PARAMETER(iqmp);
+
+    GET_PARAMETER(plaintext);
+    GET_PARAMETER(ciphertext);
+
+    RSA *key = RSA_new();
+
+    RSA_set0_key(key,
+                 BN_bin2bn(n, VECTOR_LENGTH("n"), NULL),
+                 BN_bin2bn(e, VECTOR_LENGTH("e"), NULL),
+                 BN_bin2bn(d, VECTOR_LENGTH("d"), NULL));
+    RSA_set0_factors(key,
+                     BN_bin2bn(p, VECTOR_LENGTH("p"), NULL),
+                     BN_bin2bn(q, VECTOR_LENGTH("q"), NULL));
+    RSA_set0_crt_params(key,
+                        BN_bin2bn(dmp1, VECTOR_LENGTH("dmp1"), NULL),
+                        BN_bin2bn(dmq1, VECTOR_LENGTH("dmq1"), NULL),
+                        BN_bin2bn(iqmp, VECTOR_LENGTH("iqmp"), NULL));
+
+    uint8_t _plaintext[VECTOR_LENGTH("plaintext")] = {0};
+
+    RSA_private_decrypt(RSA_size(key), ciphertext, _plaintext, key, 4);
+
+    return TEST_COMPARE(plaintext);
+}
+
+bool aes_gcm_crypto_test()
+{
+    int index = 1;
+    for (auto test_vector : aes_gcm_crypto_test_vectors)
+    {
+        if (!aes_gcm_encrypt(test_vector))
+        {
+            log_d("fail encrypt at %s case %d\n", __FUNCTION__, index);
+            continue;
+        }
+        if (!aes_gcm_decrypt(test_vector))
+        {
+            log_d("fail decrypt at %s case %d\n", __FUNCTION__, index);
+            continue;
+        }
+        index++;
+    }
+
+    if (index != aes_gcm_crypto_test_vectors.size() + 1)
+    {
+        return 0;
+    }
+
+    return 1;
```

Comment:
return value is defines as bool, use true/false

Reply:
ok